core-bundle/src/EventListener/PreviewUrlConvertListener.php

```
@@ -55,7 +58,17 @@ public function __invoke(PreviewUrlConvertEvent $event): void
                 return;
             }
 
-            $event->setUrl($page->getPreviewUrl($this->getParams($request)));
+            try {
+                $event->setUrl($page->getPreviewUrl($this->getParams($request)));
+            } catch (RouteParametersException $e) {
+                $route = $e->getRoute();
+
+                if (!$e->getPrevious() instanceof InvalidParameterException || !$route instanceof PageRoute || !$route->getPageModel()->requireItem) {
```

This is debatable. 

Swallowing any `RouteParametersException` should definitely be too loose since it catches exceptions like `RouteNotFoundException` as well.

However, swallowing any `InvalidParameterException` could be a more useful default here.
Maybe @aschempp  can share his opinion.

---

I think catching the `InvalidParameterException` is correct here. Not 100% sure though.

---

I think we can safely swallow any routing exception. If we can't generate the URL, there's just no URL for that page in the frontend, so the preview event cannot generate one. I don't think it ever makes sense to display an exception message?

---

Since the `PreviewUrlListener` only handles the preview for pages, we could get away with swallowing any `RouteParametersException`. 

However, I think this could lead to some hard to debug issues when generating the frontend preview for custom pages (no preview URL gets generated and the developer is left without an exception/trace on what went wrong). 

That's why I decided to only swallow the specific case for pages with `tl_page.requireItem` since this could occur in a correctly configured standard system.